### ISSUE TITLE
2 GPU error when training

### ISSUE THREAD

**Issue by KindleHe** [yulunzhang/RCAN#10]
I run the command to use 2 GPU for trainning:
```
CUDA_VISIBLE_DEVICES=0,1 python3 main.py --n_GPUs 2 --dir_data /root/dataset/super-resolution --model RCAN --save RCAN_BIX2_G10R20P48 --scale 2 --n_resgroups 10 --n_resblocks 20 --n_feats 64  --reset --chop --save_results --print_model --patch_size 96 2>&1 | tee $LOG
```
but get follow error:
```
Unexpected end of /proc/mounts line `overlay / overlay rw,relatime,lowerdir=/data2/docker/overlay2/l/76VGRCNKB4276UVDYJIQ4K44VI:/data2/docker/overlay2/l/MM3UKJSDI6OMZYJEQHBG5K5EBU:/data2/docker/overlay2/l/3TUQTOAGEKBLNX7DPFOKXKXUD5:/data2/docker/overlay2/l/5ZHVFRGKBYJ5MGWORLVPCB67H4:/data2/docker/overlay2/l/MGTNS2XZPIFDXQLJDPBWMZHSFF:/data2/docker/overlay2/l/NBUTJL2W2ZFDXG2JAE3Y6V4M3Z:/data2/docker/overlay2/l/WZ4AKFUGVNF4YJNSHH5XQEZVAV:/data2/docker/overlay2/l/W5VI2B4IEWSZLIUN7VC2PP3LD4:/data2/docker/overlay2/l/JBVVURDZXDPD7SAEKMXLQGX2YS:/dat'
Unexpected end of /proc/mounts line `a2/docker/overlay2/l/2ISST5GDKCNKQHI3D6LITRSPPC:/data2/docker/overlay2/l/QA7MQGMCVTSS4DQ4SS7QOEGADY:/data2/docker/overlay2/l/24BA5LASJSQBJYYNQONNE7DFOA:/data2/docker/overlay2/l/RHLGBBVVMXFSFDL666UIIDLCU6:/data2/docker/overlay2/l/ZJYKOHO5XHWZVLIG3OOX4SMJMW:/data2/docker/overlay2/l/X3VORDWXFDU2Q4IZGWZE24GOF7,upperdir=/data2/docker/overlay2/581f3545fee5eef1ebdd17aea4f9e4d4b922a18a608972a6115f2bbeec32b019/diff,workdir=/data2/docker/overlay2/581f3545fee5eef1ebdd17aea4f9e4d4b922a18a608972a6115f2bbeec32b019/work '
Unexpected end of /proc/mounts line `0 0
```
Could you please help me to find the reason?

**Comment by yulunzhang** [comment]
Hi,
I re-ran the code with multi-GPU (2, 3, or 4 GPUs) and they all passed. I used anaconda to install pytorch0.4.0. You can try with this instead of Docker.
However, I didn't use Docker to train or test the code. It seems there're some problems in /proc/mounts. If you still want to use Docker to run the code, you may refer to those links to solve your problem.
https://stackoverflow.com/questions/46138549/docker-openmpi-and-unexpected-end-of-proc-mounts-line
http://www.cloudypoint.com/Tutorials/discussion/docker-solved-docker-openmpi-and-unexpected-end-of-procmounts-line/

**Comment by yulunzhang** [comment]
Hi,
I use one Titan Xp for each model. It takes about 6 days to reach 1000 epochs. It seems that multi-gpu training cannot boost the training obviously. Each training image is converted to .npy file to save training time. 
No extra tricks were used, the results in the paper can be reproduced with my provided training scripts. Some people have reproduced the results with the code.

**Comment by KindleHe** [comment]
I see you offer some script for data converting and processing, but as a new pytorch learner, I do quite not understand how the training images being converted to .npy files?

**Comment by yulunzhang** [comment]
Hi,
Please see RCAN_TrainCode/code/data/srdata.py lines 30 to 50.

![image](https://user-images.githubusercontent.com/22829440/45600503-be7a8400-b9cb-11e8-82c3-966dd537169d.png)

**Comment by KindleHe** [comment]
Thanks, it means that I just need to `python main.py --model RCAN --save RCAN_BIX2_G10R20P48 --scale 2 --n_resgroups 10 --n_resblocks 20 --n_feats 64  --reset --chop --save_results --print_model --patch_size 96` command and the code will automatically run to the line 30 to 50 to finish the .npy generating process, am I right?

**Comment by swz30** [comment]
Hi Yulun, 
You mentioned in the above comments that "multi-gpu training cannot boost the training obviously". Why is that?

Also, could you please tell how did you end up choosing RG=10 and RCAB=20 for your model? Can I reduce any blocks to train the model faster without compromising accuracy? 

Thanks.
